<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.31">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>jumprr_steps – JUMP documentation and examples</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-e1a5c8363afafaef2c763b6775fbf3ca.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-8ef56b68f8fa1e9d2ba328e99e439f80.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-e1a5c8363afafaef2c763b6775fbf3ca.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-d1029c8c0624b678ec1882d5dbfbed62.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark-317fa9c8a93cbaa92631be92f51646c5.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../site_libs/bootstrap/bootstrap-d1029c8c0624b678ec1882d5dbfbed62.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "Search",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../howto/interactive/0_overview.html">How-To: Interactive Tools</a></li><li class="breadcrumb-item"><a href="../../howto/interactive/1_jumprr_steps.html">JUMP_rr tutorial</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">JUMP documentation and examples</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/broadinstitute/2023_12_JUMP_data_only_vignettes" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
    <div id="quarto-search" class="quarto-navigation-tool px-1" title="Search"></div>
</div>
    </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Explanations</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../explanations/data_description.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Dataset Overview</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../explanations/FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Notes and Frequently Asked Questions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../explanations/known_issues.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Known issues</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../explanations/quirks_details.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quirks and details</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Reference</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../reference/00_gene_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Available genes in JUMP</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../reference/01_chemical_query.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Available chemicals in JUMP</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../reference/computational_pipelines.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Computational Pipelines</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../reference/glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary and Acronyms</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">How-To: Notebooks</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../howto/notebooks/0_overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview of Notebooks</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../howto/notebooks/11_retrieve_profiles.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Retrieve JUMP profiles</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../howto/notebooks/12_add_metadata.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Add metadata to profiles</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../howto/notebooks/13_calculate_activity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Calculate phenotypic activity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../howto/notebooks/14_display_perturbation_images.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Display perturbation images</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../howto/notebooks/15_explore_distance_clusters.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Explore perturbation clusters</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../howto/notebooks/16_query_genes_externally.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Query genes for more info</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">How-To: Interactive Tools</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../howto/interactive/0_overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview of Interactive tools</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../howto/interactive/1_jumprr_steps.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">JUMP_rr tutorial</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../cite.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cite</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#jump_rr-tutorial" id="toc-jump_rr-tutorial" class="nav-link active" data-scroll-target="#jump_rr-tutorial">JUMP_rr tutorial</a>
  <ul class="collapse">
  <li><a href="#step-by-step-tutorial-using-jump_rr-to-answer-key-biological-questions" id="toc-step-by-step-tutorial-using-jump_rr-to-answer-key-biological-questions" class="nav-link" data-scroll-target="#step-by-step-tutorial-using-jump_rr-to-answer-key-biological-questions"><strong>Step-by-step tutorial: Using JUMP_rr to answer key biological questions</strong></a></li>
  <li><a href="#step-2-explore-the-data-to-answer-the-following-questions" id="toc-step-2-explore-the-data-to-answer-the-following-questions" class="nav-link" data-scroll-target="#step-2-explore-the-data-to-answer-the-following-questions"><strong>Step 2: Explore the data to answer the following questions:</strong></a>
  <ul class="collapse">
  <li><a href="#was-my-gene-tested-in-the-jump-collection-of-perturbations" id="toc-was-my-gene-tested-in-the-jump-collection-of-perturbations" class="nav-link" data-scroll-target="#was-my-gene-tested-in-the-jump-collection-of-perturbations"><strong>2.1. Was my gene tested in the JUMP collection of perturbations?</strong></a></li>
  <li><a href="#does-my-gene-of-interest-produce-a-morphology-phenotype-when-overexpressed-or-knocked-down-by-crispr" id="toc-does-my-gene-of-interest-produce-a-morphology-phenotype-when-overexpressed-or-knocked-down-by-crispr" class="nav-link" data-scroll-target="#does-my-gene-of-interest-produce-a-morphology-phenotype-when-overexpressed-or-knocked-down-by-crispr"><strong>2.2. Does my gene of interest produce a morphology phenotype when overexpressed or knocked down by CRISPR?</strong></a></li>
  <li><a href="#what-are-the-specific-morphology-changesfeatures" id="toc-what-are-the-specific-morphology-changesfeatures" class="nav-link" data-scroll-target="#what-are-the-specific-morphology-changesfeatures"><strong>2.2.1. What are the specific morphology changes/features?</strong></a></li>
  <li><a href="#what-other-genes-look-similar-or-anti-similar-to-my-genes" id="toc-what-other-genes-look-similar-or-anti-similar-to-my-genes" class="nav-link" data-scroll-target="#what-other-genes-look-similar-or-anti-similar-to-my-genes"><strong>2.2.2. What other genes look similar or anti-similar to my genes?</strong></a></li>
  </ul></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/broadinstitute/2023_12_JUMP_data_only_vignettes/edit/main/howto/interactive/1_jumprr_steps.md" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/broadinstitute/2023_12_JUMP_data_only_vignettes/blob/main/howto/interactive/1_jumprr_steps.md" class="toc-action"><i class="bi empty"></i>View source</a></li><li><a href="https://github.com/broadinstitute/2023_12_JUMP_data_only_vignettes/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="1_jumprr_steps.ipynb" download="1_jumprr_steps.ipynb"><i class="bi bi-journal-code"></i>Jupyter</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../howto/interactive/0_overview.html">How-To: Interactive Tools</a></li><li class="breadcrumb-item"><a href="../../howto/interactive/1_jumprr_steps.html">JUMP_rr tutorial</a></li></ol></nav></header>





<section id="jump_rr-tutorial" class="level1">
<h1>JUMP_rr tutorial</h1>
<section id="step-by-step-tutorial-using-jump_rr-to-answer-key-biological-questions" class="level2">
<h2 class="anchored" data-anchor-id="step-by-step-tutorial-using-jump_rr-to-answer-key-biological-questions"><strong>Step-by-step tutorial: Using JUMP_rr to answer key biological questions</strong></h2>
<div style="text-align: justify;">
<p>The JUMP_rr interactive <a href="https://github.com/broadinstitute/monorepo/tree/main/libs/jump_rr#quick-data-access">tools</a> allow you to explore the JUMP dataset of genetic and chemical perturbations and their effects on cell morphology features. By applying different filters, you can investigate how specific perturbations influence cells, their structures, and the strength of the changes.</p>
<p>In this tutorial, you will learn how to:</p>
<p><strong>Step 1: Understand the information displayed in the browser.</strong><br>
<strong>Step 2: Explore the data to answer the following questions:</strong></p>
<p><strong>2.1</strong>. Is my gene in the JUMP dataset?<br>
<strong>2.2.</strong> Does my gene have a morphological phenotype when overexpressed or knocked down by CRISPR?<br>
  <strong>2.2.1.</strong> If yes, what are the specific morphological changes or features?<br>
  <strong>2.2.2.</strong> What other genes look similar or anticorrelated to my gene?</p>
<section id="step-1.-understand-the-information-displayed-in-the-browser." class="level2">
<h2 class="anchored" data-anchor-id="step-1.-understand-the-information-displayed-in-the-browser."><strong>Step 1. Understand the information displayed in the browser.</strong></h2>
<p>When you open one of the interactive tables, such as the CRISPR knock-outs Feature Ranking <a href="http://broad.io/crispr_feature">table</a>, you will see an informational section explaining the available data columns for that table, and a filtering tool, followed by the data (Figure 1).</p>
<p><img src="https://zenodo.org/api/records/15252378/files/Figure1.png/content" style="max-width: 100%; height: auto; display: block;"></p>
<p><em><strong>Figure 1.</strong> <strong>Information section of the CRISPR Feature Ranking table.</strong> The red boxes highlight two key components: the first provides definitions for each column; the second one shows the tool to filter based on the content values, such as gene name, subcellular compartment or statistical significance associated with each perturbation.</em></p>
<p>Below you will find the full table (Figure 2), which displays aggregated results across replicates for each perturbation. A small triangle on one of the columns shows how the data is currently sorted and you can sort by a data column by clicking its header. Clicking again reverses the sort order. For most tables, sorting by corrected p-value from lowest to highest is a helpful view.</p>
<p><img src="https://zenodo.org/api/records/15252378/files/Figure2.png/content" style="max-width: 100%; height: auto; display: block;"></p>
<p><em><strong>Figure 2.</strong> <strong>The data section of the CRISPR Feature Ranking table, sorted by row ID.</strong> Each column can be sorted in ascending or descending order by clicking on the header. Sorting is applied to one column at a time and is based on the data type: numerical values are sorted by magnitude, while text is sorted alphabetically. The entries displayed in the Feature Ranking table are the 30 most statistically significant features for each gene (Feature Rank), and for each feature, the 30 genes with the most significant values (Gene Rank). A value of 999999 indicates an unassigned entry. How to use and interpret the Feature Ranking table is described in section 2.2.1.</em></p>
</section>
</div>
</section>
<section id="step-2-explore-the-data-to-answer-the-following-questions" class="level2">
<h2 class="anchored" data-anchor-id="step-2-explore-the-data-to-answer-the-following-questions"><strong>Step 2: Explore the data to answer the following questions:</strong></h2>
<section id="was-my-gene-tested-in-the-jump-collection-of-perturbations" class="level3">
<h3 class="anchored" data-anchor-id="was-my-gene-tested-in-the-jump-collection-of-perturbations"><strong>2.1. Was my gene tested in the JUMP collection of perturbations?</strong></h3>
<div style="text-align: justify;">
<p>The JUMP Hub includes a reference tool (Figure 3) to quickly check whether a specific gene is included in any of the catalogued genetic perturbation datasets, including the JUMP CRISPR and ORF datasets that are available via JUMP_rr tables. You can click on the ‘Available Genes’ tab in the sidebar from the <a href="https://broad.io/jump">JUMP Hub</a> page. Here, you can type the name of the gene of interest to search. Although all gene names in the tool are displayed in capital letters, the search is not case-sensitive. However, the tool does not recognize synonyms or alternative gene names, so queries must match the official gene symbols used in the datasets. The number of occurrences in multiple JUMP or JUMP-adjacent morphological profiling datasets will be displayed: “crispr” and “orf” are the JUMP knock-outs and over expression datasets, “A549”, “HeLa_DMEM” and “HeLa_HPLM” relate to the <a href="https://github.com/broadinstitute/2022_PERISCOPE">PERISCOPE</a> dataset, and <a href="https://github.com/carpenter-singh-lab/2024_LacosteHaghighi_Cell_Mislocalization">Lacoste</a> to a dataset focusing on pathogenic coding variants. This tool is especially useful when you want to collate data from multiple morphology-based datasets.</p>
<p>Note that 12,609 genes were tested in the JUMP ORF dataset (larger genes do not express well so this is roughly the maximum that could be tested) and 7,975 genes were tested in JUMP CRISPR dataset (due to resource limitations). Overall, 15,243 unique human genes were tested in one or the other datasets (or both), giving a ~75% chance your gene was tested in JUMP.</p>
<p><img src="https://zenodo.org/api/records/15252378/files/Figure3.png/content" style="max-width: 100%; height: auto; display: block;"></p>
<p><em><strong>Figure 3.</strong> <strong>Querying the glucose transporter gene SLC2A in JUMP datasets using the “Available genes” tool.</strong> Overview of the gene reference tool (red box on the left), to quickly check whether your gene of interest is available in the dataset. You can search using the full name or partial name of a gene in the search window (red box on the right). For each gene, the tool displays how many times it appears in the JUMP dataset under different perturbation types, such as CRISPR (for gene knock-out) and ORF (for overexpression), as well as across information related to the PERISCOPE (A549, HeLa_DMEM and HeLa_HPLM) and Lacoste datasets.</em></p>
</div>
</section>
<section id="does-my-gene-of-interest-produce-a-morphology-phenotype-when-overexpressed-or-knocked-down-by-crispr" class="level3">
<h3 class="anchored" data-anchor-id="does-my-gene-of-interest-produce-a-morphology-phenotype-when-overexpressed-or-knocked-down-by-crispr"><strong>2.2. Does my gene of interest produce a morphology phenotype when overexpressed or knocked down by CRISPR?</strong></h3>
<div style="text-align: justify;">
<p>To answer this question, you will start by exploring the <strong>CRISPR knock-out dataset</strong> to identify genes whose loss affects the phenotype of interest. Then, you will use the ORF overexpression dataset to contrast results with gene overexpression. Of the 15,243 genes tested in JUMP, 68% (10,352) yielded a detectable phenotype (statistically significant “phenotypic activity”), that is, an image-based profile with a signal distinct from negative controls by ORF, CRISPR, or both.</p>
<p>To begin exploring these morphological phenotypes, navigate to the <strong>“How to Guide → Interactive Tools”</strong>. Then, click on the <strong>“JUMP_rr tools”</strong> option, this will take you to the repository where the interactive tables are hosted. Once in the repository, scroll down to the <strong>“Quick Data Access”</strong> section. There, you’ll find links to the CRISPR, ORF, and Compound datasets. Click on the table of interest in the <strong>“Feature Ranking”</strong> column to see the morphological features associated with each perturbation. You can also directly access the feature ranking tables using the following short links:</p>
<ul>
<li><a href="http://broad.io/crispr_feature">CRISPR knock-out feature table</a><br>
</li>
<li><a href="http://broad.io/orf_feature">ORF overexpression feature table</a><br>
</li>
<li><a href="http://broad.io/feature">Compound feature table</a></li>
</ul>
<p><strong><em>Search for your gene in the CRISPR dataset.</em></strong></p>
<p>Click on the “Feature Ranking” link for the CRISPR knock-down dataset to open a table displaying the morphological features associated with each gene perturbation. From there, you can use the filtering tools to search for your gene of interest and sort the data by criteria such as statistical significance or subcellular compartment. This view helps you determine whether a gene perturbation induces detectable phenotypic changes, and reveals the specific morphological features altered in the cells.</p>
<p>Let us explore data related to the SLC2A2 gene. This gene encodes a glucose transporter primarily expressed in organs such as liver, pancreas and intestine. It plays a key role in glucose sensing and homeostasis. Given its involvement in metabolic processes, changes in SLC2A2 activity or expression could have significant phenotypic consequences that may be captured in image-based profiling data.</p>
<p>To look for this you can use the Filter Tool, which helps narrow down the dataset based on specific search criteria. When you open the filter menu, a list of all available column names will appear (Figure 4A). In this case, you would select the “Perturbation” column, use the default operator “=”, and enter “SLC2A2” in the search field to display only the rows associated with that gene (Figure 4B).</p>
<p>If you’re unsure of the exact name used in the dataset, we recommend make your search case-insensitive by changing the default filter operator from “=” to “like”, and entering a pattern that partially matches a text string, such as “SLC2A%”, since “%” act as a wildcard representing any characters (Figure 4C).</p>
<p>You can also search for alternative names commonly used in the literature. To do this, select the “Synonyms” column, change the filter operator to “contains” or “like”, and enter an alternative name such as GLUT2, or another known synonym of your compound of interest (Figure 4D).</p>
<p><img src="https://zenodo.org/api/records/15252378/files/Figure4.png/content" style="max-width: 100%; height: auto; display: block;"></p>
<p><em><strong>Figure 4.</strong> <strong>Common filtering methods that can be used in the Feature Ranking table</strong>. Panel A shows the full list of columns available for custom filtering, including metadata such as Corrected p-value, Compartment, and others that support more advanced queries beyond gene-level filtering. B and C illustrate how the number of entries returned depends on the filter type: a broader search using partial names (e.g., “SLC2A%”) yields more entries than an exact match (e.g., “SLC2A2”), which can help identify related genes within the same family. D highlights the utility of the Synonyms column, which maps alternative gene names used in the literature, especially useful for genes with multiple aliases or legacy names not directly listed under the Perturbation column.</em></p>
</div>
<p>The table below describes the available logical operators (Table1).</p>
<table class="caption-top table">
<colgroup>
<col style="width: 50%">
<col style="width: 50%">
</colgroup>
<thead>
<tr class="header">
<th>Operator</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>=</td>
<td>Selects results that exactly match the specified value.</td>
</tr>
<tr class="even">
<td>!=</td>
<td>Excludes results that exactly match the specified value.</td>
</tr>
<tr class="odd">
<td>contains</td>
<td>Searches for values that include a specific string within the text.</td>
</tr>
<tr class="even">
<td>ends with</td>
<td>Selects values that end with a specific string.</td>
</tr>
<tr class="odd">
<td>starts with</td>
<td>Selects values that start with a specific string.</td>
</tr>
<tr class="even">
<td>&gt;</td>
<td>Selects numeric values greater than the specified value.</td>
</tr>
<tr class="odd">
<td>≥</td>
<td>Selects numeric values greater than or equal to the specified value.</td>
</tr>
<tr class="even">
<td>&lt;</td>
<td>Selects numeric values less than the specified value.</td>
</tr>
<tr class="odd">
<td>≤</td>
<td>Selects numeric values less than or equal to the specified value.</td>
</tr>
<tr class="even">
<td>like</td>
<td>Searches for patterns that partially match a text string (e.g., “SLC%”).</td>
</tr>
<tr class="odd">
<td>not like</td>
<td>Excludes results that partially match a specific pattern.</td>
</tr>
<tr class="even">
<td>in</td>
<td>Selects rows where the column value is within a specified list of values.</td>
</tr>
<tr class="odd">
<td>not in</td>
<td>Excludes rows where the column value is within a specified list of values.</td>
</tr>
<tr class="even">
<td>is blank</td>
<td>Filters rows where the value is an empty string or space — the field exists but has no visible content.</td>
</tr>
<tr class="odd">
<td>is not blank</td>
<td>Filters rows where the field contains visible content (text, numbers, etc.), excluding empty or space-only values.</td>
</tr>
</tbody>
</table>
<div style="text-align: justify;">
<p><em><strong>Table 1.</strong> <strong>Filtering operators can be used to refine values displayed in the dataset</strong>. Include or exclude rows based on specific conditions, such as matching exact values, checking for patterns in text, or evaluating numerical comparisons.</em></p>
<p>You can combine multiple filters. If you want to filter for several values within the same column for example, multiple compartments or gene names, select the appropriate column, choose the <strong>‘in’</strong> operator, and enter a list of values separated by commas (e.g., ‘Cytoplasm, Nuclei’) (Figure 5). Once the filters are applied, you can copy the filtered table by clicking the JSON hyperlink (Figure 5, red box). To export the content, press Ctrl+A to select all, then Ctrl+C to copy it, and finally paste it into a text editor. Note that export as a “CSV” option is not yet available.</p>
<p><img src="https://zenodo.org/api/records/15252378/files/Figure5.png/content" style="max-width: 100%; height: auto; display: block;"></p>
<p><em><strong>Figure 5.</strong> <strong>Combined filters and export options that can be used to explore and integrate gene-related data.</strong> Example of combined filters applied to display gene-related data for SLC2A2 limited to the “Cells” and “Nuclei” compartments. The red box highlights available export options, such as JSON, which can streamline the integration of filtered results into external analysis pipelines or visualization tools.</em></p>
<p>After applying the filter for the <strong>SLC2A2</strong> perturbation, you can evaluate whether the perturbation is significantly different to the negative control condition by examining the <strong>Corrected p-value</strong> column. The Corrected p-value represents the statistical significance of how distinctive the overall morphological profile of a perturbation is when compared to the negative control (after false discovery rate correction). In this case, the corrected p-value is <strong>0.00018</strong> (all rows display the same value in that column, because this value is computed per perturbation, not per feature). This value (p-value&lt;0.05) indicates a significant difference from the control condition, suggesting that knocking out <strong>SLC2A2</strong> induces a strong and consistent morphological effect (Figure 6).</p>
<p><img src="https://zenodo.org/api/records/15252378/files/Figure6.png/content" style="max-width: 100%; height: auto; display: block;"></p>
<p><em><strong>Figure 6.</strong> <strong>Corrected p-value that can be used to prioritize perturbations with significant morphological effects</strong> The user interface displays the “Corrected p-value” for SLC2A2, which is calculated at the perturbation level. As a result, all rows associated with the SLC2A2 perturbation share the same value (0.00018). When exploring multiple perturbations, sorting by Corrected p-value (or filtering based on it) can help prioritize those with the most statistically significant morphological effects, making it a useful strategy for identifying relevant information. The rows are sorted in ascending/descending order by clicking the name column header.</em></p>
</div>
</section>
<section id="what-are-the-specific-morphology-changesfeatures" class="level3">
<h3 class="anchored" data-anchor-id="what-are-the-specific-morphology-changesfeatures"><strong>2.2.1. What are the specific morphology changes/features?</strong></h3>
<div style="text-align: justify;">
<p>In this section you will be able to identify which features are the most statistically significant within a perturbation. In this filtered view of the SLC2A2 perturbation, all values in the <strong>Feature significance</strong> column are <strong>0.0,</strong> which corresponds to values rounded to five decimal places and indicates that each listed morphological feature is statistically significant when compared to the control condition (Figure 7). If you’d like more information about the features extracted by CellProfiler, such as how they’re named, you can check out the “<em>How Measurements are Named</em>” section in <a href="https://cellprofiler-manual.s3.amazonaws.com/CellProfiler-4.0.4/help/output_measurements.html">CellProfiler Handbook</a>.</p>
<p><img src="https://zenodo.org/api/records/15252378/files/Figure7.png/content" style="max-width: 100%; height: auto; display: block;"></p>
<p><em><strong>Figure 7.</strong> <strong>Top features that can be sorted and prioritized based on Feature Significance for the SLC2A2 perturbation in the JUMP CRISPR dataset.</strong> Top four features from the filtered CRISPR dataset are sorted in ascending order based on the Feature Significance column (highlighted with a red box). All p-values in this column are displayed as 0.0, which reflects values approximated to five decimal places (i.e., &lt; 0.00001). Because many features share these low values, we recommend using <strong>Feature Rank</strong> to prioritize features, as it provides their relative importance for each perturbation.</em></p>
<p><strong><em>Search for your gene in ORF dataset.</em></strong></p>
<p>To explore whether overexpressing SLC2A2 produces a morphological phenotype, you will repeat the same filtering process in the ORF overexpression dataset’s Feature Ranking table, link <a href="http://broad.io/orf_feature">here</a>. Using the Filter Tool, you will select the Synonyms column, keep the operator as “=”, and enter “SLC2A2” in the search field.</p>
<p>After filtering, you can examinate the <strong>Corrected p-value</strong> column. In this case, the value is <strong>0.09482</strong>, which is above the 0.05 threshold, indicating that overexpressing <em>SLC2A2</em> does not produce a statistically significant morphological change. This contrasts with the CRISPR knock-out dataset, where the loss of <em>SLC2A2</em> resulted in a strong and significant phenotypic effect.</p>
<p>However, in the case of SLC2A2, it is still possible to identify individual morphological features that are statistically significant, even if the overall perturbation does not show significant changes compared to the control. In this filtered view, a few features have <strong>Feature Significance</strong> values below 0.05, indicating meaningful differences relative to the control condition (Figure 8).These p-values are corrected for multiple hypothesis testing.</p>
<p><img src="https://zenodo.org/api/records/15252378/files/Figure8.png/content" style="max-width: 100%; height: auto; display: block;"></p>
<p><em><strong>Figure 8.</strong> <strong>Top features that can be sorted and prioritized based on Feature Significance for the SLC2A2 perturbation in the JUMP ORF dataset.</strong> The top features from the filtered dataset are sorted in ascending order based on the JUMP ORF Feature Ranking table, based on feature significance (per-feature p-values), ranked in increasing order, indicating statistically significant differences compared to the control condition. The Channel and Suffix columns are empty because these features solely come from an object mask and not a fluorescence image. The rows are sorted in ascending/descending order by clicking the name column header.</em></p>
</div>
</section>
<section id="what-other-genes-look-similar-or-anti-similar-to-my-genes" class="level3">
<h3 class="anchored" data-anchor-id="what-other-genes-look-similar-or-anti-similar-to-my-genes"><strong>2.2.2. What other genes look similar or anti-similar to my genes?</strong></h3>
<div style="text-align: justify;">
<p>Before exploring which perturbations produce the most similar or anti-similar image-based profile (i.e., features that are high in one are lower in the other one), it’s important to understand the broader context and potential of image-based profiling.</p>
<p>Morphological profiles can cluster genes into biologically meaningful groups, uncovering known pathways and revealing functional connections, such as a regulatory interaction between pathways (Rohban et al.&nbsp;2017). This clustering is based on the similarity of high-dimensional morphological features extracted from images after gene overexpression or CRISPR knock-out. Comparing these profiles systematically can also identify <em>anti-correlated</em> perturbations, such as cases where one perturbation produces the opposite phenotype of another (Rohban et al., 2022). This is particularly relevant for therapeutic discovery, as finding antagonistic morphological profiles may highlight candidate compounds or genes that <em>suppress</em> an undesired phenotype.</p>
<p>The “matches” table thus quantifies correlation (or anticorrelation) of profiles and is a powerful resource to uncover potential functional relationships between genes, explore unknown gene roles, or identify suppressors or enhancers within a biological pathway (Chandrasekaran et al., 2024; Ramezani et al., 2025). These insights can help prioritize follow-up experiments and hypothesis generation, especially when analyzing genes or perturbations with poorly characterized functions.</p>
<p>To analyze the SLC2A2 perturbation in the CRISPR matches <a href="http://broad.io/crispr">table</a>, begin by opening the <strong>Filter</strong> menu. From the list of available columns, select <strong>“Perturbation”</strong>, choose the <strong>“like”</strong> operator, and type <strong>“SLC2A2”</strong> in the search field. Then, sort the <strong>“Perturbation-Match Similarity”</strong> column in <strong>descending order</strong> by clicking on the name in that column to view the perturbations with the strongest phenotypic similarity at the top (Figure 9). The similarity scores range theoretically from -1 (strong anti-similarity) to 1 (strong similarity), although values above 0.8 and below -0.8 are rarely seen, and values above 0.3 and below -0.3 tend to indicate a reliable “match”. Typically, scores above 0.3 and below -0.3 indicate a reliable “match” or “anti-match.” As observed in Section 2.2 <em>Search for your gene in the CRISPR dataset</em>, SLC2A2 knock-out displays a strong phenotypic effect, and when compared to other perturbations, a wide range of similarity scores is observed, from a maximum of 0.621, strong phenotypic similarity - a “match”, to a minimum of -0.513, strong phenotypic anti-similarity - an “anti-match” (Figure 9, red box). <strong>The “Perturbation example images” and “Match example image”</strong> shown in the table are randomly sampled from replicates across the experiment (not necessarily from the same plate or batch as the match), providing an unbiased visual representation of each condition. The “Match resources column links to external databases where additional information about the matching/anti-matching gene can be explored (Figure 9, blue box). These images are also useful to check for technical artifacts (e.g., out-of-focus fields, staining issues) or abnormal cell counts (too few or too many cells), which may affect the interpretation of the morphological phenotype.</p>
<p>Now, let’s analyze the SLC2A2 overexpression in the ORF matches table to explore the matches there. Start by opening the <strong>Filter</strong> menu. From the list of available columns, select <strong>“Perturbation”</strong>, choose the <strong>“like”</strong> operator, and enter <strong>“SLC2A2”</strong> in the search field. Next, sort the <strong>“Perturbation-Match Similarity”</strong> column in descending order by clicking on its header as you did for the CRISPR table. In this case, the range of similarity scores is narrower than in the CRISPR dataset, spanning approximately from <strong>0.27</strong> to <strong>-0.223</strong>. This limited dynamic range may reflect the fact that <em>SLC2A2</em> overexpression induces a weaker morphological phenotype compared to its knock-out as you explored in the Section 2.2.1. <em>Search for your gene in the ORF dataset.</em></p>
<p>Taken together, these comparisons provide a starting point for hypothesis generation and pathway exploration, particularly when integrating multiple types of perturbations.</p>
<p><img src="https://zenodo.org/api/records/15252378/files/Figure9.png/content" style="max-width: 100%; height: auto; display: block;"></p>
<p><em><strong>Figure 9.</strong> <strong>Comparison between the selected perturbation, SLC2A2, in the CRISPR Match table.</strong> After filtering for SCL2A2, the top 4 strongest matches are shown. The Perturbation-Match Similarity column (red box) indicates how similar or anti-similar each perturbation is to SLC2A2 (click on column name to sort it). The Match Resources column (blue box) provides external links to databases with additional information about the genes being compared. The merged images of all Cell Painting channels for SLC2A2 and its matched perturbation can be inspected in the “Perturbation example image” and “Match example image” columns, allowing visual assessment of their morphological signatures to visual comparison of morphological signatures and potential technical artifacts.</em></p>
<p><strong>References:</strong></p>
<p>Mohammad Hossein Rohban, Shantanu Singh, Xiaoyun Wu, Julia B Berthet, Mark-Anthony Bray, Yashaswi Shrestha, Xaralabos Varelas, Jesse S Boehm, Anne E Carpenter (2017) Systematic morphological profiling of human gene and allele function via Cell Painting eLife 6:e24060</p>
<p>Mohammad H. Rohban, Ashley M. Fuller, Ceryl Tan, Jonathan T. Goldstein, Deepsing Syangtan, Amos Gutnick, Ann DeVine, Madhura P. Nijsure, Megan Rigby, Joshua R. Sacher, Steven M. Corsello, Grace B. Peppler, Marta Bogaczynska, Andrew Boghossian, Gabrielle E. Ciotti, Allison T. Hands, Aroonroj Mekareeya, Minh Doan, Jennifer P. Gale, Rik Derynck, Thomas Turbyville, Joel D. Boerckel, Shantanu Singh, Laura L. Kiessling, Thomas L. Schwarz, Xaralabos Varelas, Florence F. Wagner, Ran Kafri, T.S. Karin Eisinger-Mathason, Anne E. Carpenter (2022) Virtual screening for small-molecule pathway regulators by image-profile matching. <em>Cell Systems</em>, 13(9): 724–736.e9.</p>
<p>Srinivas Niranj Chandrasekaran, Eric Alix, John Arevalo, Adriana Borowa, Patrick J. Byrne, William G. Charles, Zitong S. Chen, Beth A. Cimini, Boxiong Deng, John G. Doench, Jessica D. Ewald, Briana Fritchman, Colin J. Fuller, Jedidiah Gaetz, Amy Goodale, Marzieh Haghighi, Yu Han, Zahra Hanifehlou, Holger Hennig, Desiree Hernandez, Christina B. Jacob, Tim James, Tomasz Jetka, Alexandr A. Kalinin, Ben Komalo, Maria Kost-Alimova, Tomasz Krawiec, Brittany A. Marion, Glynn Martin, Nicola Jane McCarthy, Lisa Miller, Arne Monsees, Nikita Moshkov, Alán F. Muñoz, Arnaud Ogier, Magdalena Otrocka, Krzysztof Rataj, David E. Root, Francesco Rubbo, Simon Scrace, Douglas W. Selinger, Rebecca A. Senft, Peter Sommer, Amandine Thibaudeau, Sarah Trisorus, Rahul Valiya Veettil, William J. Van Trump, Sui Wang, Michał Warchoł, Erin Weisbart, Amélie Weiss, Michael Wiest, Agata Zaremba, Andrei Zinovyev, Shantanu Singh, Anne E. Carpenter (2024) Morphological map of under- and over-expression of genes in human cells. bioRxiv. https://doi.org/10.1101/2024.12.02.624527</p>
<p>Ramezani, M., Weisbart, E., Bauman, J. et al.&nbsp;A genome-wide atlas of human cell morphology. Nat Methods 22, 621–633 (2025). https://doi.org/10.1038/s41592-024-02537-7</p>
</div>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/shntnu\.github\.io\/jump_hub\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/broadinstitute/2023_12_JUMP_data_only_vignettes/edit/main/howto/interactive/1_jumprr_steps.md" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/broadinstitute/2023_12_JUMP_data_only_vignettes/blob/main/howto/interactive/1_jumprr_steps.md" class="toc-action"><i class="bi empty"></i>View source</a></li><li><a href="https://github.com/broadinstitute/2023_12_JUMP_data_only_vignettes/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>